src/main/java/roomescape/api/common/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,42 @@
+package roomescape.api.common;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.client.HttpClientErrorException;
+import roomescape.domain.reservation.ReservationIsExists;
+import roomescape.domain.reservation.ReservationNotFound;
+import roomescape.domain.reservation.ReservationScheduleNotValid;
+import roomescape.domain.theme.ThemeIsUsed;
+import roomescape.domain.theme.ThemeNotFound;
+import roomescape.domain.time.ReservationTimeIsUsed;
+import roomescape.domain.time.ReservationTimeNotFound;
+
+@RestControllerAdvice
+public class GlobalExceptionHandler {
+
+  @ExceptionHandler(HttpClientErrorException.class)
+  public ResponseEntity<String> catchClientError(HttpClientErrorException e) {
+    return new ResponseEntity<>(e.getMessage(), e.getStatusCode());
+  }
+
+  @ExceptionHandler(ReservationScheduleNotValid.class)
+  @ResponseStatus(HttpStatus.BAD_REQUEST)
+  public String checkValidError(RuntimeException e) {
```

Comment:
```suggestion
  public String checkValidError(ReservationScheduleNotValid e) {
```
가능하면 ExceptionHandler에서 잡는 예외와 인자로 들어오는 예외를 맞추는 것이 좋습니다.
- [[Spring docs] Exceptions](https://docs.spring.io/spring-framework/reference/web/webmvc/mvc-controller/ann-exceptionhandler.html)